cranelift/assembler/src/imm.rs

```diff
@@ -53,18 +125,44 @@ impl Imm32 {
     pub fn value(&self) -> u32 {
         self.0
     }
-}
-
-impl std::fmt::Display for Imm32 {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        if self.0 < 10 {
-            write!(f, "${:x}", self.0)
+    pub fn to_string(&self, extend: Extension) -> String {
+        if matches!(extend, Extension::SignExtendQuad) {
```

Comment:
How about a helper function to simplify and deduplicate?

```rust
match extend {
  SignExtendQuad => possibly_print_as_hex(self.0 as i32 as i64),
  SignExtendLong => possibly_print_as_hex(self.0 as i32),
  ...
}
```

I guess we would need the function `possibly_print_as_hex` to be generic over `T` since there are different types involved plus whatever numeric `ops` traits are needed to do `<` and `>=`.